pkgs/README.md

```diff
@@ -384,7 +384,9 @@ All versions of a package _must_ be included in `all-packages.nix` to make sure
 * `meta.license` must be set and match the upstream license.
   * If there is no upstream license, `meta.license` should default to `lib.licenses.unfree`.
   * If in doubt, try to contact the upstream developers for clarification.
-* `meta.mainProgram` must be set when appropriate.
+* `meta.mainProgram` must be set when there is any main executable for the package in `$bin/bin` or `$out/bin`.
+  * Packages like `e2fsprogs` that have multiple executables, none of which can be considered the main program, should not set `meta.mainProgram`.
+  * Always use a hardcoded string (don't use `pname`, for example).
```

Comment:
Thanks for putting together this PR!  It is a nice improvement!

Somewhat of a nitpick, but it might be nice to give the following as well:

- A simple example of a package that only has one executable, where `meta.mainProgrm` should be set.  This just helps give a easy-to-understand baseline for what `meta.mainProgram` is mainly for.

- An example of a package where the attribute name is different from the main executable name (like `ripgrep` / `rg`).  This helps drive the point home that you may have to think a little bit about what `meta.mainProgram` should be, and it is not always just the attribute name.

- An example of a package that is just a library and doesn't provide executables.  It should be made perfectly clear that `meta.mainProgram` should not be set.  This helps clarify that `meta.mainProgram` isn't necessary on all derivations.

- An example of a package that is primarily a library, but does also provide executables that most users wouldn't be interested in using (in which case it should be made clear that `meta.mainProgram` should not be set).  This helps clarify that sometimes it is not clear whether or not `meta.mainProgram` should be set, and it is sometimes at the maintainer's discretion.

Although maybe this sort of information is a better fit for the Nixpkgs manual.

Reply:
Thanks, do the updates address your request?

Reply:
Looks good, thanks!